mongo/datadog_checks/mongo/api.py

```diff
@@ -95,7 +95,11 @@ def _authenticate(self):
     def _get_deployment_type(self):
         # getCmdLineOpts is the runtime configuration of the mongo instance. Helpful to know whether the node is
         # a mongos or mongod, if the mongod is in a shard, if it's in a replica set, etc.
-        options = self['admin'].command("getCmdLineOpts")['parsed']
+        try:
+            options = self['admin'].command("getCmdLineOpts")['parsed']
+        except Exception:
```

Comment:
```suggestion
        except Exception as e:
            self.log.debug("Unable to run `getCmdLineOpts`, got: %s", str(e))
```

Should the check log that it has detected a nontypical-deployment type (Alibaba AsparaDB)?